exec/shell_test.go

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2024-Present Defense Unicorns
+
+package exec
+
+import (
+	"testing"
+)
+
+func TestGetOSShell(t *testing.T) {
+	type test struct {
+		shell     ShellPreference
```

Comment:
rename symbols for `ShellPreference` type to be `shellPref` or `pref` to increase clarity